src/MyMarketManager.Data.Processing/WorkItemProcessingEngine.cs

```diff
@@ -0,0 +1,270 @@
+using System.Threading.Channels;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace MyMarketManager.Data.Processing;
+
+/// <summary>
+/// Central processing engine that orchestrates work item processing using System.Threading.Channels.
+/// Features:
+/// - Each registered handler fetches its own work items
+/// - Bounded channels prevent memory issues
+/// - Fair scheduling prevents starvation
+/// - Single background service for all processors
+/// </summary>
+public class WorkItemProcessingEngine
+{
+    private readonly IServiceProvider _serviceProvider;
+    private readonly ILogger<WorkItemProcessingEngine> _logger;
+    private readonly List<IWorkItemHandlerRegistration> _registrations = new();
+    private volatile bool _initialized = false;
+
+    public WorkItemProcessingEngine(
+        IServiceProvider serviceProvider,
+        ILogger<WorkItemProcessingEngine> logger)
+    {
+        _serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+    }
+
+    /// <summary>
+    /// Initializes the engine with all registered handlers.
+    /// Should be called once during startup.
+    /// </summary>
+    public void Initialize()
+    {
+        if (_initialized)
+            return;
+
+        WorkItemProcessingExtensions.ConfigureEngine(_serviceProvider, this);
```

Comment:
I am not a fan of this. Is there no way to use some IOptions and then in each add handler we do a services.Configure.

I want my engine to be ready when it is instantiaged and not have to have some external system call init.